### PR TITLE
homework1: vector2d implementation

### PR DESCRIPTION
**Pull request by IlyaKarpuk**

### CODE REVIEW

**Review by maciekgajewski — COMMENTED** [review]
good, but incomplete

File: course4/homework1/test.cc [hunk 1]
```diff
@@ -23,5 +23,15 @@ int main()
...
 
+	print(Vector2d{1, 0});
+	print(Vector2d{0, 1});
+	print(Vector2d{-1, 0});
+	print(Vector2d{0, -1});
+	print(Vector2d{1, 2});
+	print(Vector2d{-1, 2});
+	print(Vector2d{1, -2});
+	print(Vector2d{-1, -2});
+	print(Vector2d::FromPolarCoordinates(1, 180));
```
Comment (maciekgajewski) [inline course4/homework1/test.cc]: This is a manual test; automated test would be better

File: course4/homework1/vector2d.h [hunk 2]
```diff
@@ -1,9 +1,71 @@
...
+				return 0; // not really defined
+
+			if (mY > 0)
+				return 90;
+
+			return 270;
+		}
+
+		if (mX < 0)
+			return atan(mY / mX) / AngleToRadian + 180;
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: Use `std::atan`, `std::cos` etc

File: course4/homework1/vector2d.h [hunk 3]
```diff
@@ -1,9 +1,71 @@
...
+	double GetX() const { return mX;}
+	void SetX(double value) { mX = value; }
+
+	double GetY() const { return mY;}
+	void SetY(double value) { mY = value; }
+	
+	double GetRadius() const { return sqrt(mX*mX + mY*mY); }
+	double GetAngle() const
+	{
+		if (mX == 0)
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: This is `std::atan2`

File: course4/homework1/vector2d.h [hunk 4]
```diff
@@ -1,9 +1,71 @@
 #pragma once
 
+#include <math.h>
 #include <ostream>
+#include <stdexcept>
 
-struct Vector2d
+const double AngleToRadian = M_PI / 180;
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: make all  your constants `static`

File: course4/homework1/vector2d.h [hunk 5]
```diff
@@ -1,9 +1,71 @@
...
+			return 270;
+		}
+
+		if (mX < 0)
+			return atan(mY / mX) / AngleToRadian + 180;
+
+		if (mY < 0)
+			return atan(mY / mX) / AngleToRadian + 360;
+
+		return atan(mY / mX) / AngleToRadian;
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: I would expect the function to return the angle in radians, like all the other math functions do

**Review by maciekgajewski — COMMENTED** [review]
Make the tests automated, don't bother converting radians to degrees, use qualified functions.

Consider making the constructor private

File: course4/homework1/test.cc (hunk 1 above)
Comment (IlyaKarpuk) [reply course4/homework1/test.cc]: what do you mean by automated tests? unit testing with gtest for example?

Comment (maciekgajewski) [reply course4/homework1/test.cc]: Just extending the suite of simple, assert-based tests is OK

File: course4/homework1/test.cc [hunk 6]
```diff
@@ -4,24 +4,34 @@
 #include <cassert>
 #include <iostream>
 
+bool AreEqual(Vector2d x, Vector2d y) { return x == y; }
```
Comment (maciekgajewski) [inline course4/homework1/test.cc:7]: This function has no point

File: course4/homework1/test.cc [hunk 7]
```diff
@@ -4,24 +4,34 @@
...
 
-	assert((a+b == Vector2d{4, 13}));
+	assert(a+b == Vector2d::FromCartesianCoordinates(4, 13));
 
 	std::stringstream s;
 	s << a;
 
 	assert(s.str() == "(5, 6)");
 
+	assert(AreEqual(
```
Comment (maciekgajewski) [inline course4/homework1/test.cc:23]: You can just use the operator==

File: course4/homework1/vector2d.h [hunk 8]
```diff
@@ -1,9 +1,59 @@
...
+	double GetRadius() const { return std::sqrt(mX*mX + mY*mY); }
+	double GetAngle() const
+	{
+		return std::atan2(mY, mX);
+	}
+
+	static Vector2d FromPolarCoordinates(double radius, double angle)
+	{
+		if (radius < 0)
+			throw std::runtime_error("'radius' cannot be negative");
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h:30]: why?

**Review by maciekgajewski — COMMENTED** [review]
mosty good. The "Are Equal'
 is not needed

**Pull request closed without merging**